backend/src/graphql/resolver/TransactionResolver.test.ts

```diff
@@ -0,0 +1,318 @@
+            },
+          }),
+        ).toEqual(
+          expect.objectContaining({
+            errors: [new GraphQLError('memo text is too long (255 characters maximum)')],
+          }),
+        )
+      })
+
+      it('logs the error thrown', () => {
+        expect(logger.error).toBeCalledWith('memo text is too long: memo.length=256 > 255')
+      })
+    })
+
+    describe('memo text is too short', () => {
+      it('throws an error', async () => {
+        expect(
+          await mutate({
+            mutation: sendCoins,
+            variables: {
+              email: 'peter@lustig.de',
+              amount: 100,
+              memo: 'test',
+            },
+          }),
+        ).toEqual(
+          expect.objectContaining({
+            errors: [new GraphQLError('memo text is too short (5 characters minimum)')],
+          }),
+        )
+      })
+
+      it('logs the error thrown', () => {
+        expect(logger.error).toBeCalledWith('memo text is too short: memo.length=4 < 5')
+      })
+    })
+
+    describe('user has not enough GDD', () => {
+      it('throws an error', async () => {
+        expect(
+          await mutate({
+            mutation: sendCoins,
+            variables: {
+              email: 'peter@lustig.de',
+              amount: 100,
+              memo: 'testing',
+            },
+          }),
+        ).toEqual(
+          expect.objectContaining({
+            errors: [new GraphQLError(`user hasn't enough GDD or amount is < 0`)],
+          }),
+        )
+      })
+
+      it('logs the error thrown', () => {
+        expect(logger.error).toBeCalledWith(
+          `user hasn't enough GDD or amount is < 0 : balance=null`,
+        )
+      })
+    })
+  })
+
+  describe('transaction correct', () => {
+    it('sends the coins', async () => {
+      // make Peter Lustig Admin
+      const peter = await User.findOneOrFail({ id: user[0].id })
+      peter.isAdmin = new Date()
+      await peter.save()
+
+      // create contribution as user bob
+      const contribution = await mutate({
+        mutation: createContribution,
+        variables: { amount: 1000, memo: 'testing', creationDate: new Date().toISOString() },
+      })
+
+      // login as admin
+      await query({ query: login, variables: peterData })
+
+      // confirm the contribution
+      await mutate({
+        mutation: confirmContribution,
+        variables: { id: contribution.data.createContribution.id },
+      })
+
+      // login as bob again
+      await query({ query: login, variables: bobData })
+
+      expect(
+        await mutate({
+          mutation: sendCoins,
+          variables: {
+            email: 'peter@lustig.de',
+            amount: 100,
```

Comment:
This could be flaky. If the CI needs too much time, the decay applied could become relevant and the test fails because there are not enough GDD left. Take something like 50

Reply:
okay, anyway the user had 1000GDD, but no problem on setting that to 50